build/common.ps1

```diff
@@ -315,16 +317,34 @@ Function Get-LatestVisualStudioRoot() {
     Error-Log 'Cannot find an instance of Visual Studio 2017 or newer' -Fatal
 }
 
+Function Get-VSVersion() {
+    if (-not $VSVersion) {
+        $VSVersion = $DefaultVSVersion
+    }
+    return $VSVersion
+}
+
+Function Get-VSMajorVersion() {
+    $vsVersion = Get-VSVersion
+    $vsMajorVersion = $vsVersion.Split('.')[0]
+    return $vsMajorVersion
+}
+
 Function Get-MSBuildRoot {
     param(
         [switch]$Default
     )
+
+    $vsMajorVersion = Get-VSMajorVersion
+
     # Willow install workaround
     if (-not $Default) {
+
         # Find version 15.0 or newer
-        if (Test-Path Env:\VS150COMNTOOLS) {
+        $CommonToolsVar = "Env:VS${vsMajorVersion}0COMNTOOLS"
```

Comment:
Personally I'd let this here determine the version. 
Look for either 15/16 version of the env variable. 
If either exists use it.

Reply:
As currently I'm not sure if the following scenario is handled. 
I have a:
- Dev16 Preview
- Dev16 stable

The preview is the higher version. 

I am in the Dev16 stable developer command prompt, and I want to build using that same stable msbuild version (we've had cases where the previews have bugs that prevent us from building). 

Right now I don't see that handled, as Dev15 is the default. 

In Dev15, this would have been autopicked. 

tldr; 

I think if you are in a valid compatbile developer command prompt (Dev15/Dev16), you should just build with that.

Reply:
This is handled by Get-VSVersion. 
Made a small change in the retrieval of the env var to make this more consistent.